content/docs/rfcs/43/README.md

```diff
@@ -0,0 +1,347 @@
+in order to frequently and deterministically change the `messageNametag` of messages exchanged during the pairing and transfer phase - 
+ideally, at each message exchanged. 
+
+Given the proposed construction,
+the `ctsInbound` and `ctsOutbound` secrets can be used to iteratively generate the `messageNametag` field of Waku payloads 
+for inbound and outbound messages, respectively. 
+
+The derivation of `messageNametag` should be deterministic only for communicating devices 
+and independent from message content, 
+otherwise lost messages will prevent computing the next message nametag. 
+A possible approach consists in computing the `n`-th `messageNametag` as `H( ctsInbound || n)`, 
+where `n` is serialized as `uint64`.
+
+In this way, sender's and recipient's devices 
+can keep updated a buffer of `messageNametag` to sieve 
+while listening to messages sent over `/{application-name}/{application-version}/wakunoise/1/sessions-{shard-id}/` (i.e., the next 50 not yet seen). 
+They will then be able to further identify if one or more messages were eventually lost 
+or not-yet-delivered during the communication.
+This approach brings also the advantage that 
+communicating devices can efficiently identify encrypted messages addressed to them.
+
+We note that since the `ChaChaPoly` cipher used to encrypt messages supports *additional data*, 
+an encrypted payload can be further authenticated by passing the `messageNametag` as additional data to the encryption/decryption routine. 
+In this way, an attacker would be unable to craft an authenticated Waku message 
+even in case the currently used symmetric encryption key is compromised, 
+unless `ctsInbound`, `ctsOutbound` or the `messageNametag` buffer lists were compromised too.
+
+# Security/Privacy Considerations
+
+### Assumptions
+
+- The attacker is active, i.e. can interact with both devices `A` and `B` by sending messages over `contentTopic`.
+
+- The attacker has access to the QR code, that is knows the ephemeral key `eB`, the commitment `H(sB||r)` and the `contentTopic` exposed by the device `B`.
+
+- Devices `A` and `B` are considered trusted (otherwise the attacker will simply exfiltrate the relevant information from the attacked device).
+
+- As common for Noise, we assume that ephemeral keys cannot be compromised, while static keys might be later compromised. However, we enforce in the pairing phase extra security mechanisms (i.e. use of commitments for static keys) that will prevent some attacks possible when ephemeral keys are weak or get compromised.
+ 
+### Rationale
+
+- The device `B` exposes a commitment to its static key `sB` because:
+    - if the private key of `eB` is weak or gets compromised, an attacker can impersonate `B` by sending in message `c.` to device `A` his own static key and successfully complete the pairing phase. Note that being able to compromise `eB` is not contemplated by our security assumptions.
```

Comment:
The beginning and the ending of this sentence contradict i.e., "if the private key of `eB` is weak or gets compromised" and "Note that being able to compromise `eB` is not contemplated by our security assumptions". I'd suggest only touching on the parts that the protocol is designed for.